docs/api-reference/carto/data-sources.md

```diff
@@ -0,0 +1,220 @@
+# Data Sources
+
+To ease interacting with the CARTO platform, the CARTO deck.gl module includes a number of functions, which simplify the use of fetching your data from CARTO. At a high level these can be thought of as wrappers around the browsers `fetch` function, except that rather than passing a URL, options that specify the data in the CARTO are used.
+
+### Overview
+
+The data source functions are a compact way to describe the data that you want to fetch. For example, to fetch a table from a data warehouse from the connection `carto_dw`:
+
+```js
+import {vectorTableSource} from '@deck.gl/carto';
+const data = vectorTableSource({
+  accessToken: 'XXX',
+  connectionName: 'carto_dw',
+  tableName: 'carto-demo-data.demo_tables.chicago_crime_sample',
+})
+```
+
+### Promise API
+
+All data source functions return a Promise, which can be resolved to obtain the actual Tilejson. However, as the [core deck.gl Layer prop supports Promises](../core/layer#data), it is often not necessary to resolve or await the Promise and the data source can be directly passed to the data prop:
+
+```jsx
+import {H3TileLayer, h3TilesetSource} from '@deck.gl/carto';
+new H3TileLayer({
+  data: h3TilesetSource({
+    accessToken: 'XXX',
+    connectionName: 'carto_dw',
+    tableName: 'carto-demo-data.demo_tables.h3_data'
+  }),
+
+  getFillColor: d => d.properties.color
+});
+```
+
+### Types
+
+All the data source functions are fully typed, to aid in providing the correct parameters and working correctly with the return value.
+
+### Caching
+
+The dataSource functions have an internal cache, which avoids fetching data from the server if the parameters have not changed. Thus they can be used, for example, in React `render()` functions without needing memoization.
+
+### Available Data Sources
+
+All data source functions take the following global options:
+
+```ts
+type SourceOptions = {
+  accessToken: string;
+  connectionName: string;
+  apiBaseUrl?: string;
+  clientId?: string;
+  headers?: Record<string, string>;
+  mapsUrl?: string;
+};
+```
+
+In addition, the following options are supported on each source:
+
+#### h3TableSource
+
+```ts
+type H3TableSourceOptions = {
+  aggregationExp: string;
+  aggregationResLevel?: number;
+  columns?: string[];
+  spatialDataColumn?: string;
+  tableName: string;
+}
+```
+
+#### h3QuerySource
+
+```ts
+type H3QuerySourceOptions = {
+  aggregationExp: string;
+  aggregationResLevel?: number;
+  columns?: string[];
+  spatialDataColumn?: string;
+  sqlQuery: string;
+  queryParameters: QueryParameters;
+}
+```
+
+#### h3TilesetSource
+
+```ts
+type H3TilesetSourceOptions = {
+  tableName: string;
+}
+```
+
+#### vectorTableSource
+
+```ts
+type vectorTableSourceOptions = {
+  columns?: string[];
+  spatialDataColumn?: string;
+  tableName: string;
+}
+```
+
+#### vectorQuerySource
```

Comment:
Could you move the vectorXXXSource to the top as they're the most used? Order: vector, h3, quadbin, raster